From 162aecc95c62cf855082ee79550542ddeb8bf37a Mon Sep 17 00:00:00 2001
From: Even Xu <Feng.Xu@freescale.com>
Date: Thu, 29 Dec 2011 14:07:00 +0800
Subject: [PATCH 264/449] ENGR00171036 V4L2 Capture AutoRun test Enable

- Enable V4L2 Capture autorun test.
- Add more test case for V4L2 Capture

Signed-off-by: Even Xu <b21019@freescale.com>
---
 test/mxc_v4l2_test/autorun-v4l2.sh | 170 +++++++++++++++++++++++++++++++++----
 1 file changed, 154 insertions(+), 16 deletions(-)

diff --git a/test/mxc_v4l2_test/autorun-v4l2.sh b/test/mxc_v4l2_test/autorun-v4l2.sh
index 6422f59..2322094 100755
--- a/test/mxc_v4l2_test/autorun-v4l2.sh
+++ b/test/mxc_v4l2_test/autorun-v4l2.sh
@@ -17,17 +17,30 @@ else
 fi
 
 # devnode test
+# for camera
 if [ $TEST_CAMERA = 1 ]; then
-if [ "$(platform)" = "IMX35_3STACK" ]; then
-insmod ipu_prp_enc
-insmod ipu_prp_vf_sdc
-insmod ipu_prp_vf_sdc_bg
-insmod ipu_still
-insmod ov2640_camera
-insmod mxc_v4l2_capture
-fi
+	if [ "$(platform)" = "IMX35_3STACK" ]; then
+		modprobe ov2640_camera
+		modprobe mxc_v4l2_capture
+	fi
+	if ([ "$(platform)" = IMX51 ] || [ "$(platform)" = IMX53 ] \
+	|| [ "$(platform)" = IMX6 ]); then
+		if [ "$(platform)" = "IMX51" ]; then
+			modprobe ov3640_camera
+		fi
+		if [ "$(platform)" = "IMX53" ]; then
+			modprobe ov5642_camera
+		fi
+		if [ "$(platform)" = "IMX6" ]; then
+			modprobe ov5640_camera
+		fi
+		modprobe mxc_v4l2_capture
+	fi
 check_devnode "/dev/video0"
 fi
+
+# devnode test
+# for display
 if [ "$(platform)" != IMX25_3STACK ]; then
 check_devnode "/dev/video16"
 fi
@@ -140,18 +153,143 @@ if [ $TEST_CAMERA = 1 ]; then
 
 # V4L2 Capture Tests
 if [ "$(platform)" = IMX25_3STACK ]; then
-run_testcase "./csi_v4l2_overlay.out -t 10 -w 640 -h 480 -fr 30"
+	run_testcase "./csi_v4l2_overlay.out -t 10 -w 640 -h 480 -fr 30"
 else
-run_testcase "./mxc_v4l2_overlay.out -iw 640 -ih 480 -ow 240 -oh 320 -r 4 -fr 30 -fg -t 10"
-run_testcase "./mxc_v4l2_overlay.out -iw 640 -ih 480 -ow 240 -oh 320 -r 4 -fr 30 -t 10"
 
-for ROT in 0 1 2 3 4 5 6 7; do
-	run_testcase "./mxc_v4l2_overlay.out -iw 640 -ih 480 -ow 240 -oh 184 -r $ROT -fr 30 -fg -t 5"
-done
+for MODE in 0 1 2 3 4 5 6; do
+	for RATO in 15 30; do
+		if [ "$(platform)" = "IMX51" ]; then
+			# ov3640_camera driver not support 30fps@mode3
+			if ([ $MODE = 3 ] && [ $RATO = 30 ]); then
+				continue
+			fi
+
+			if [ $MODE = 0 ]; then
+				WIDTH=640
+				HIGHT=480
+			elif [ $MODE = 1 ]; then
+				WIDTH=320
+				HIGHT=240
+			elif [ $MODE = 2 ]; then
+				WIDTH=1024
+				HIGHT=768
+			elif [ $MODE = 3 ]; then
+				WIDTH=2048
+				HIGHT=1536
+			elif [ $MODE = 4 ]; then
+				WIDTH=720
+				HIGHT=480
+			elif [ $MODE = 5 ]; then
+				WIDTH=720
+				HIGHT=576
+			else
+				continue
+			fi
+		fi
+		if [ "$(platform)" = "IMX53" ]; then
+			if [ $RATO = 15 ]; then
+				if ([ $MODE = 0 ] || [ $MODE = 1 ] || [ $MODE = 2 ] || [ $MODE = 3 ] || [ $MODE = 4 ]); then
+					continue
+				elif [ $MODE = 5 ]; then
+					WIDTH=1920
+					HIGHT=1080
+				else
+					WIDTH=2592
+					HIGHT=1944
+				fi
+			fi
+			if [ $RATO = 30 ]; then
+				if [ $MODE = 0 ]; then
+					WIDTH=640
+					HIGHT=480
+				elif [ $MODE = 1 ]; then
+					WIDTH=320
+					HIGHT=240
+				elif [ $MODE = 2 ]; then
+					WIDTH=720
+					HIGHT=480
+				elif [ $MODE = 3 ]; then
+					WIDTH=720
+					HIGHT=576
+				elif [ $MODE = 4 ]; then
+					WIDTH=1280
+					HIGHT=720
+				else
+					continue
+				fi
+			fi
+		fi
+		if [ "$(platform)" = "IMX6" ]; then
+			# ov5640_camera driver not support
+			# 15fps@moder0 mode1 mode2 mode3 mode4 mode5
+			# 30fps@mode6
+			if ([ $MODE = 6 ] && [ $RATO = 30 ]); then
+				continue
+			fi
+
+			if [ $RATO = 15 ]; then
+				if ([ $MODE = 0 ] || [ $MODE = 1 ] || [ $MODE = 2 ] || [ $MODE = 3 ] || [ $MODE = 4 ] || [ $MODE = 5 ]); then
+					continue
+				fi
+			fi
 
-for POS in 0 4 8 16 32 64 128; do
-	run_testcase "./mxc_v4l2_overlay.out -iw 640 -ih 480 -ot $POS -ol $POS -ow 80 -oh 60 -fr 30 -fg -t 5"
+			if [ $MODE = 0 ]; then
+				WIDTH=640
+				HIGHT=480
+			elif [ $MODE = 1 ]; then
+				WIDTH=320
+				HIGHT=240
+			elif [ $MODE = 2 ]; then
+				WIDTH=720
+				HIGHT=480
+			elif [ $MODE = 3 ]; then
+				WIDTH=720
+				HIGHT=576
+			elif [ $MODE = 4 ]; then
+				WIDTH=1280
+				HIGHT=720
+			elif [ $MODE = 5 ]; then
+				WIDTH=1920
+				HIGHT=1080
+			else
+				WIDTH=2592
+				HIGHT=1944
+			fi
+
+		fi
+
+		echo ==== v4l2 overlay base test: $WIDTH X $HIGHT @ $RATO ====
+		for RUNTIME in 5 10 20 40 70 120; do
+			echo ==== foreground runtime: $RUNTIME ====
+			run_testcase "./mxc_v4l2_overlay.out -iw $WIDTH -ih $HIGHT -ow 640 -oh 480 -fr $RATO -m $MODE -t $RUNTIME -fg"
+			echo ==== background runtime: $RUNTIME ====
+			run_testcase "./mxc_v4l2_overlay.out -iw $WIDTH -ih $HIGHT -ow 640 -oh 480 -fr $RATO -m $MODE -t $RUNTIME"
+		done
+
+		echo ==== v4l2 overlay rotation test: $WIDTH X $HIGHT @ $RATO ====
+		for ROT in 0 1 2 3 4 5 6 7; do
+			echo ==== foreground rotation: $ROT ====
+			run_testcase "./mxc_v4l2_overlay.out -iw $WIDTH -ih $HIGHT -ow 640 -oh 480 -fr $RATO -m $MODE -r $ROT -fg -t 5"
+			echo ==== background rotation: $ROT ====
+			run_testcase "./mxc_v4l2_overlay.out -iw $WIDTH -ih $HIGHT -ow 640 -oh 480 -fr $RATO -m $MODE -r $ROT -t 5"
+		done
+
+		echo ==== v4l2 overlay position test: $WIDTH X $HIGHT @ $RATO ====
+		for POS in 0 4 8 16 32 64 128; do
+			echo ==== foreground position: $POS ====
+			run_testcase "./mxc_v4l2_overlay.out -iw $WIDTH -ih $HIGHT -ot $POS -ol $POS -ow 640 -oh 480 -fr $RATO -m $MODE -fg -t 5"
+			echo ==== background position: $POS ====
+			run_testcase "./mxc_v4l2_overlay.out -iw $WIDTH -ih $HIGHT -ot $POS -ol $POS -ow 640 -oh 480 -fr $RATO -m $MODE -t 5"
+		done
+
+		echo ==== v4l2 capture test: $WIDTH X $HIGHT @ $RATO ====
+		for FCOUNT in 5 10 15 20 25 30; do
+			echo ==== save frame count: $FCOUNT ====
+			run_testcase "./mxc_v4l2_capture.out -$WIDTH -ih $HIGHT -ow 640 -oh 480 -fr $RATO -m $MODE -c $FCOUNT test.yuv"
+		done
+	done
 done
+
 fi
 
 fi
-- 
1.8.0

